### PR TITLE
[exporter/loadbalancing] Allow interval and timeout to be customized

### PR DESCRIPTION
**Pull request by jvilhuber**
**Description:** <Describe what has changed.>
Adding resolving interval and timeout to the configuration.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10199

**Testing:** unit tests pass, and I compiled and deployed this in my local testbed with a 30 seconds/ 5 second timeout test.

**Documentation:** new config options documented in the README.md

### CODE REVIEW

File: exporter/loadbalancingexporter/config.go
Comment (TylerHelmuth) [inline exporter/loadbalancingexporter/config.go:46]: Can the setting of the default port be moved here?

File: exporter/loadbalancingexporter/resolver_dns.go
Comment (TylerHelmuth) [inline exporter/loadbalancingexporter/resolver_dns.go:111]: holdover from debugging?

Comment (jvilhuber) [reply exporter/loadbalancingexporter/resolver_dns.go:111]: I thought it might be good to leave in to confirm things are working at the debug level. But I can certainly take it out.

File: exporter/loadbalancingexporter/config.go
Comment (jvilhuber) [reply exporter/loadbalancingexporter/config.go:46]: Good idea. Will do.

Comment (jvilhuber) [reply exporter/loadbalancingexporter/config.go:46]: While implementing this, I noticed that I can't seem to configure this without a port anyway: there's no warning or error during load, but during execution, after the backend has been discovered (the metrics indicate an expected num_backends of 1), I see
```
{"level":"warn","ts":1653367580.393693,"caller":"zapgrpc/zapgrpc.go:191","msg":"[core] [Channel #13 SubChannel #14] grpc: addrConn.createTransport failed to connect to {\n  \"Addr\": \"10.1.0.79:4317\",\n  \"ServerName\": \"10.1.0.79:4317\",\n  \"Attributes\": null,\n  \"BalancerAttributes\": null,\n  \"Type\": 0,\n  \"Metadata\": null\n}. Err: connection error: desc = \"transport: Error while dialing dial tcp 10.1.0.79:4317: operation was canceled\"","grpc_log":true}
```
If I explicitly add a `Port: "4317"` to the Dns resolver, then it all works. Very odd.

Can someone confirm that this actually works without an explicit Port?

Comment (TylerHelmuth) [reply exporter/loadbalancingexporter/config.go:46]: Looking quickly through the exporter's code I also don't see DNSResolver setting a default port despite the readme saying it should.  Might be a bug?

Comment (jvilhuber) [reply exporter/loadbalancingexporter/config.go:46]: The reason I was testing this and slightly concerned is because of metrics labels: I was trying to see if in the default case the endpoint label is JUST "hostname" (or IP). In all the cases I was able to test, the endpoint label was "hostname:port". I'd hate to create a backwards compatibility issue by changing the labels in the current default case by added the port to the label (this breaking some people's dashboards or alerts or whatever).
But I haven't been able to confirm this one way or another since it doesn't appear to work in my tests..

File: exporter/loadbalancingexporter/resolver_dns.go
Comment (jvilhuber) [reply exporter/loadbalancingexporter/resolver_dns.go:111]: So is it convention to not have debug logs? Or are you just concerned there's not much information in the log?

Comment (TylerHelmuth) [reply exporter/loadbalancingexporter/resolver_dns.go:111]: I'm fine with leaving it

File: exporter/loadbalancingexporter/config.go
Comment (TylerHelmuth) [reply exporter/loadbalancingexporter/config.go:46]: Can you open an issue then details your findings? The readme says there should be a default port in DNSResolver but I don't see that happening. The solution that gets discussed in the issue may be to update the README.

Comment (jvilhuber) [reply exporter/loadbalancingexporter/config.go:46]: Issue opened: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10321

Comment (jvilhuber) [reply exporter/loadbalancingexporter/config.go]: Back to your original request: "Can the setting of the default port be moved here?"
In trying to implement this, I'm having to go deep into tests and behavior changes. All the unit tests simply test on "endpoint-1" and "endpoint-2", which (at least for the dns-resolver) is now no longer a valid value (the port will be required, since we would add it by default). Then, by adding the port to all those tests, lots of tests that assume a hashring-lookup-result fail, because we get a different endpoint-result when searching for a specific TraceId now (I didn't get too far into why).
I'd suggest that someone with more knowledge of the code pursue this thread of explicitly adding the default port value into the DNSResolver structure.

Comment (TylerHelmuth) [reply exporter/loadbalancingexporter/config.go]: Yes we'll save it for a different PR.

**Review by TylerHelmuth — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
@jvilhuber please sign the CLA 👍

**Review by pmm-sumo — APPROVED** [review]
Other than the CLA missing LGTM :)

**Comment by jvilhuber** [comment]
> CLA

Yea, I'm working on it. Sorry for the delay.

**Comment by TylerHelmuth** [comment]
@jvilhuber any news on the CLA?

**Comment by jvilhuber** [comment]
> @jvilhuber any news on the CLA?

Still waiting on my company to approve. Sorry about that.

**Comment by jpkrohling** [comment]
What's the state of this PR?

**Comment by jvilhuber** [comment]
> What's the state of this PR?

Still waiting for me (or rather my company) to sign the CLA. It's winding its way through the layers. Hopefully soon.

**Pull request merged**
